home/views/api/utils.py

```diff
@@ -58,3 +60,26 @@ def validate_request_json(
             break
 
     return response
+
+def require_authn(func: Callable[[View, Any, Any], HttpResponse]):
+    """Decorator for Django View methods to require authn.
+
+    Checks if the request's user is authenticated. If not, returns a 401 HttpResponse.
+    Otherwise, calls the decorated method.
+
+    Parameters
```

Comment:
We can also use in some built-in middleware available, which re-directs to the login page while this one doesn't.